lib/OpenQA/WebAPI/Controller/API/V1/JobSettings.pm

```diff
@@ -10,37 +10,58 @@ use Mojo::Base 'Mojolicious::Controller', -signatures;
 
 Filters jobs based on a single setting key/value pair.
 
+  openqa-cli api job_settings/jobs key="ISSUES[]" value=39911
   openqa-cli api job_settings/jobs key="*_TEST_ISSUES" list_value=39911
   openqa-cli api job_settings/jobs key="*" list_value=39911
 
 =item C<key>
 
-The setting key to filter by. It accepts a string or a glob of a job variable.
-Some variables are not stored and they can not be used.
+The setting key to filter by. Some variables are not stored and they can not be
+used.
+
+=item C<value>
+
+Returns all jobs where the value of the setting specified by C<key> is C<value>.
+
+As this query might find many jobs the results are limited by the server-side
+setting C<{misc_limits}{generic_default_limit}>. This limit can be increased
+using the C<limit> paremter (up to C<{misc_limits}{generic_max_limit}>).
 
 =item C<list_value>
 
-The value to match for the given key. This can be a string without special characters.
+Returns all jobs where the value of the setting specified by C<key> contains a
+comma-separated list that in turn contains C<list_value>. This can be a string
+without special characters. When C<list_value> is used, C<key> might contain an
+asterisks for globbing.
+
+Use of C<list_value> is expensive so the search is limited via the server-side
+setting C<{misc_limits}{job_settings_max_recent_jobs}>.
 
 =item Returns
 
-On success, returns an array of the matched job ids. The results rely on
-C<{misc_limits}{job_settings_max_recent_jobs}>.
+On success, returns an array of the matched job IDs.
 
 =back
 
 =cut
 
 sub jobs ($self) {
     my $validation = $self->validation;
-    $validation->required('key')->like(qr/^[\w\*]+$/);
-    $validation->required('list_value')->like(qr/^\w+$/);
+    $validation->required('key')->like(qr/^[\w\*\[\]]+$/);
+    $validation->optional('value')->like(qr/^\w+$/);
+    $validation->optional('list_value')->like(qr/^\w+$/);
+    $validation->optional('limit')->num(0);
     return $self->reply->validation_error({format => 'json'}) if $validation->has_error;
 
     my $key = $validation->param('key');
+    my $value = $validation->param('value');
     my $list_value = $validation->param('list_value');
-    my $jobs = $self->schema->resultset('JobSettings')->jobs_for_setting({key => $key, list_value => $list_value});
-    $self->render(json => {jobs => $jobs});
+    my $limit = $validation->param('limit');
+    return $self->render(json => {error => 'either "value" or "list_value" needs to be specified'}, status => 400)
+      unless defined($value) ^^ defined($list_value);
```

Comment:
That's a syntax error (should be `^`). But I would also prefer `xor` here because ["Binary "^" returns its operands XORed together bit by bit."](https://perldoc.perl.org/perlop#Bitwise-Or-and-Exclusive-Or) and it's not intended for checking for truthiness, and `xor` is also a bit more readable.
```suggestion
      unless (defined($value) xor defined($list_value));
```

Reply:
I was using the `^^` operator as documented on https://perldoc.perl.org/perlop#C-style-Logical-Or,-Xor,-and-Defined-Or.

> Binary "^^" performs a logical XOR operation.

It worked in my local tests. Maybe it has only been introduced in a recent version of Perl.

I avoided `xor` because tidy did some very ugly reformatting when it was used. I guess we'll have to live with that.